src/Adapter/MSTest.TestAdapter/Execution/TestMethodInfo.cs

```diff
@@ -420,11 +427,33 @@ private TestResult ExecuteInternal(object?[]? arguments, ExecutionContext? execu
                     if (RunTestInitializeMethod(_classInstance, result, ref executionContext, timeoutTokenSource))
                     {
                         hasTestInitializePassed = true;
-                        FixtureMethodRunner.RunOnContext(executionContext, () =>
+                        var tcs = new TaskCompletionSource<object?>();
```

Comment:
The core of the fix is here.
`InvokeAsSynchronousTask` would have previously blocked the UI thread, preventing the continuation from running.